src/polygon/src/polygon.cpp

```diff
@@ -637,10 +656,15 @@ class EarClip {
       const float d2 = glm::dot(diff, diff);
       if (d2 < minD2 && vert->pos.y * above > start->pos.y * above &&
           above * glm::determinant(glm::mat2(left, offset)) > -precision_ &&
-          above * glm::determinant(glm::mat2(offset, right)) > -precision_ &&
-          !vert->IsConvex(precision_)) {
```

Comment:
This was the cause of the incorrect bridge ordering. Convexity is not the right check - instead we need to check that our start vert is on the inside of the connector vert's two edges.